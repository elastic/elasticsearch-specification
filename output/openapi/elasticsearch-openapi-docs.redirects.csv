async-search-submit-1,async-search-submit
bulk-1,bulk
cat-aliases-1,cat-aliases
cat-allocation-1,cat-allocation
cat-component-templates-1,cat-component-templates
cat-count-1,cat-count
cat-fielddata-1,cat-fielddata
cat-indices-1,cat-indices
cat-ml-data-frame-analytics-1,cat-ml-data-frame-analytics
cat-ml-datafeeds-1,cat-ml-datafeeds
cat-ml-jobs-1,cat-ml-jobs
cat-ml-trained-models-1,cat-ml-trained-models
cat-recovery-1,cat-recovery
cat-segments-1,cat-segments
cat-shards-1,cat-shards
cat-snapshots-1,cat-snapshots
cat-templates-1,cat-templates
cat-thread-pool-1,cat-thread-pool
cat-transforms-1,cat-transforms
ccr-get-auto-follow-pattern-1,ccr-get-auto-follow-pattern
clear-scroll-1,clear-scroll
cluster-get-component-template-1,cluster-get-component-template
cluster-health-1,cluster-health
cluster-state-1,cluster-state
cluster-state-2,cluster-state
cluster-stats-1,cluster-stats
connector-put-1,connector-put
count-1,count
enrich-get-policy-1,enrich-get-policy
field-caps-1,field-caps
fleet-msearch-1,fleet-msearch
health-report-1,health-report
ilm-get-lifecycle-1,ilm-get-lifecycle
index-1,index
indices-analyze-1,indices-analyze
indices-clear-cache-1,indices-clear-cache
indices-data-streams-stats-1,indices-data-streams-stats
indices-delete-alias-1,indices-delete-alias
indices-exists-alias-1,indices-exists-alias
indices-flush-1,indices-flush
indices-forcemerge-1,indices-forcemerge
indices-get-alias-1,indices-get-alias
indices-get-alias-2,indices-get-alias
indices-get-alias-3,indices-get-alias
indices-get-data-stream-1,indices-get-data-stream
indices-get-field-mapping-1,indices-get-field-mapping
indices-get-index-template-1,indices-get-index-template
indices-get-mapping-1,indices-get-mapping
indices-get-settings-1,indices-get-settings
indices-get-settings-2,indices-get-settings
indices-get-settings-3,indices-get-settings
indices-get-template-1,indices-get-template
indices-put-alias-1,indices-put-alias
indices-put-settings-1,indices-put-settings
indices-recovery-1,indices-recovery
indices-refresh-1,indices-refresh
indices-resolve-cluster-1,indices-resolve-cluster
indices-rollover-1,indices-rollover
indices-segments-1,indices-segments
indices-shard-stores-1,indices-shard-stores
indices-simulate-template-1,indices-simulate-template
indices-stats-1,indices-stats
indices-stats-2,indices-stats
indices-stats-3,indices-stats
indices-validate-query-1,indices-validate-query
inference-delete-1,inference-delete
inference-get-1,inference-get
inference-get-2,inference-get
inference-inference-1,inference-inference
inference-put-1,inference-put
inference-update-1,inference-update
ingest-get-geoip-database-1,ingest-get-geoip-database
ingest-get-ip-location-database-1,ingest-get-ip-location-database
ingest-get-pipeline-1,ingest-get-pipeline
ingest-simulate-1,ingest-simulate
logstash-get-pipeline-1,logstash-get-pipeline
mget-1,mget
migration-deprecations-1,migration-deprecations
ml-delete-expired-data-1,ml-delete-expired-data
ml-delete-forecast-1,ml-delete-forecast
ml-explain-data-frame-analytics-1,ml-explain-data-frame-analytics
ml-get-buckets-1,ml-get-buckets
ml-get-calendars-1,ml-get-calendars
ml-get-categories-1,ml-get-categories
ml-get-data-frame-analytics-1,ml-get-data-frame-analytics
ml-get-data-frame-analytics-stats-1,ml-get-data-frame-analytics-stats
ml-get-datafeed-stats-1,ml-get-datafeed-stats
ml-get-datafeeds-1,ml-get-datafeeds
ml-get-filters-1,ml-get-filters
ml-get-job-stats-1,ml-get-job-stats
ml-get-jobs-1,ml-get-jobs
ml-get-memory-stats-1,ml-get-memory-stats
ml-get-model-snapshots-1,ml-get-model-snapshots
ml-get-trained-models-1,ml-get-trained-models
ml-get-trained-models-stats-1,ml-get-trained-models-stats
ml-preview-data-frame-analytics-1,ml-preview-data-frame-analytics
ml-preview-datafeed-1,ml-preview-datafeed
msearch-1,msearch
msearch-template-1,msearch-template
mtermvectors-1,mtermvectors
nodes-hot-threads-1,nodes-hot-threads
nodes-info-1,nodes-info
nodes-info-2,nodes-info
nodes-info-3,nodes-info
nodes-reload-secure-settings-1,nodes-reload-secure-settings
nodes-stats-1,nodes-stats
nodes-stats-2,nodes-stats
nodes-stats-3,nodes-stats
nodes-stats-4,nodes-stats
nodes-stats-5,nodes-stats
nodes-usage-1,nodes-usage
nodes-usage-2,nodes-usage
nodes-usage-3,nodes-usage
put-script-1,put-script
rank-eval-1,rank-eval
render-search-template-1,render-search-template
rollup-get-jobs-1,rollup-get-jobs
rollup-get-rollup-caps-1,rollup-get-rollup-caps
scroll-1,scroll
search-1,search
search-application-get-behavioral-analytics-1,search-application-get-behavioral-analytics
search-shards-1,search-shards
search-template-1,search-template
searchable-snapshots-cache-stats-1,searchable-snapshots-cache-stats
searchable-snapshots-clear-cache-1,searchable-snapshots-clear-cache
searchable-snapshots-stats-1,searchable-snapshots-stats
security-change-password-1,security-change-password
security-create-service-token-1,security-create-service-token
security-get-privileges-1,security-get-privileges
security-get-privileges-2,security-get-privileges
security-get-role-1,security-get-role
security-get-role-mapping-1,security-get-role-mapping
security-get-service-accounts-1,security-get-service-accounts
security-get-service-accounts-2,security-get-service-accounts
security-get-user-1,security-get-user
security-has-privileges-1,security-has-privileges
simulate-ingest-1,simulate-ingest
slm-get-lifecycle-1,slm-get-lifecycle
snapshot-get-repository-1,snapshot-get-repository
snapshot-status-1,snapshot-status
snapshot-status-2,snapshot-status
tasks-cancel-1,tasks-cancel
termvectors-1,termvectors
transform-get-transform-1,transform-get-transform
transform-preview-transform-1,transform-preview-transform
watcher-ack-watch-1,watcher-ack-watch
watcher-execute-watch-1,watcher-execute-watch
watcher-stats-1,watcher-stats
