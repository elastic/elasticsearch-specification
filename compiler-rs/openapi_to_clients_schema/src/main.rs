// Licensed to Elasticsearch B.V. under one or more contributor
// license agreements. See the NOTICE file distributed with
// this work for additional information regarding copyright
// ownership. Elasticsearch B.V. licenses this file to you under
// the Apache License, Version 2.0 (the "License"); you may
// not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

use std::collections::BTreeSet;
use std::path::PathBuf;

use openapi_to_clients_schema::openapi::OpenAPI;
use tracing::{info, Level};
use tracing_subscriber::FmtSubscriber;

fn main() -> anyhow::Result<()> {
    let subscriber = FmtSubscriber::builder().with_max_level(Level::TRACE).finish();
    tracing::subscriber::set_global_default(subscriber)?;

    // let file = "../../ent-search/swagger/v1/enterprise-search.json";
    // let file = "../../ent-search/swagger/v1/workplace-search.json";
    let file = "./openapi_to_clients_schema/fixtures/workplace-search.json";

    let src = PathBuf::from(file);
    let dest = src.with_extension("schema.json");

    info!("Loading OpenAPI from {file}");

    let file = std::fs::File::open(file)?;

    let json_deser = &mut serde_json::Deserializer::from_reader(file);

    // Track unused fields, to find any additional stuff the OpenAPI model would miss
    let mut unused = BTreeSet::new();

    let open_api: openapiv3::OpenAPI = serde_ignored::deserialize(json_deser, |path| {
        unused.insert(path.to_string());
    })?;

    if !unused.is_empty() {
        println!("Unused fields in the OpenAPI schema: {:?}", unused);
    }

    let sch_json = openapi_to_clients_schema::generate(&OpenAPI(open_api))?;
    let output = std::fs::File::create(dest)?;
    serde_json::to_writer_pretty(output, &sch_json)?;

    Ok(())
}
