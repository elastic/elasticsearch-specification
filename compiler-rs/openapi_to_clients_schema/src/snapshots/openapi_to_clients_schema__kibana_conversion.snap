---
source: openapi_to_clients_schema/src/main.rs
expression: sch_json
---
{
  "endpoints": [
    {
      "name": "_global.system_status",
      "description": "Get Kibana's current status",
      "request": {
        "namespace": "_global",
        "name": "SystemStatusRequest"
      },
      "requestBodyRequired": false,
      "response": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponse"
      },
      "urls": [
        {
          "path": "/api/status",
          "methods": [
            "GET"
          ]
        }
      ],
      "responseMediaType": [
        "application/json"
      ]
    }
  ],
  "types": [
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusRedactedResponse"
      },
      "description": "A minimal representation of Kibana's operational status.",
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "status",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusRedactedResponseStatus"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusRedactedResponseStatus"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "overall",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusRedactedResponseStatusOverall"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusRedactedResponseStatusOverall"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "level",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusRedactedResponseStatusOverallLevel"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusRedactedResponseStatusOverallLevel"
      },
      "description": "Service status levels as human and machine readable values.",
      "isOpen": false,
      "members": [
        {
          "name": "available"
        },
        {
          "name": "degraded"
        },
        {
          "name": "unavailable"
        },
        {
          "name": "critical"
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponse"
      },
      "description": "Kibana's operational status as well as a detailed breakdown of plugin statuses indication of various loads (like event loop utilization and network traffic) at time of request.",
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "metrics",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseMetrics"
            }
          },
          "required": true
        },
        {
          "name": "name",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        },
        {
          "name": "status",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatus"
            }
          },
          "required": true
        },
        {
          "name": "uuid",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        },
        {
          "name": "version",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseVersion"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseMetrics"
      },
      "description": "Metric groups collected by Kibana.",
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "collection_interval_in_millis",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "double"
            }
          },
          "required": true
        },
        {
          "name": "elasticsearch_client",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseMetricsElasticsearchClient"
            }
          },
          "required": true
        },
        {
          "name": "last_updated",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseMetricsElasticsearchClient"
      },
      "description": "Current network metrics of Kibana's Elasticsearch client.",
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "totalActiveSockets",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "double"
            }
          },
          "required": true
        },
        {
          "name": "totalIdleSockets",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "double"
            }
          },
          "required": true
        },
        {
          "name": "totalQueuedRequests",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "double"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatus"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "core",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCore"
            }
          },
          "required": true
        },
        {
          "name": "overall",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusOverall"
            }
          },
          "required": true
        },
        {
          "name": "plugins",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusPlugins"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCore"
      },
      "description": "Statuses of core Kibana services.",
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "elasticsearch",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearch"
            }
          },
          "required": true
        },
        {
          "name": "savedObjects",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjects"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearch"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "detail",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "documentationUrl",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "level",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearchLevel"
            }
          },
          "required": true
        },
        {
          "name": "meta",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearchMeta"
            }
          },
          "required": true
        },
        {
          "name": "summary",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearchLevel"
      },
      "description": "Service status levels as human and machine readable values.",
      "isOpen": false,
      "members": [
        {
          "name": "available"
        },
        {
          "name": "degraded"
        },
        {
          "name": "unavailable"
        },
        {
          "name": "critical"
        }
      ]
    },
    {
      "kind": "type_alias",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreElasticsearchMeta"
      },
      "description": "An unstructured set of extra metadata about this service.",
      "type": {
        "kind": "instance_of",
        "type": {
          "namespace": "_builtins",
          "name": "Dictionary"
        },
        "generics": [
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "UserDefined"
            }
          }
        ]
      }
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjects"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "detail",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "documentationUrl",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "level",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjectsLevel"
            }
          },
          "required": true
        },
        {
          "name": "meta",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjectsMeta"
            }
          },
          "required": true
        },
        {
          "name": "summary",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjectsLevel"
      },
      "description": "Service status levels as human and machine readable values.",
      "isOpen": false,
      "members": [
        {
          "name": "available"
        },
        {
          "name": "degraded"
        },
        {
          "name": "unavailable"
        },
        {
          "name": "critical"
        }
      ]
    },
    {
      "kind": "type_alias",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusCoreSavedObjectsMeta"
      },
      "description": "An unstructured set of extra metadata about this service.",
      "type": {
        "kind": "instance_of",
        "type": {
          "namespace": "_builtins",
          "name": "Dictionary"
        },
        "generics": [
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "UserDefined"
            }
          }
        ]
      }
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusOverall"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "detail",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "documentationUrl",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "level",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusOverallLevel"
            }
          },
          "required": true
        },
        {
          "name": "meta",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusOverallMeta"
            }
          },
          "required": true
        },
        {
          "name": "summary",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusOverallLevel"
      },
      "description": "Service status levels as human and machine readable values.",
      "isOpen": false,
      "members": [
        {
          "name": "available"
        },
        {
          "name": "degraded"
        },
        {
          "name": "unavailable"
        },
        {
          "name": "critical"
        }
      ]
    },
    {
      "kind": "type_alias",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusOverallMeta"
      },
      "description": "An unstructured set of extra metadata about this service.",
      "type": {
        "kind": "instance_of",
        "type": {
          "namespace": "_builtins",
          "name": "Dictionary"
        },
        "generics": [
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "UserDefined"
            }
          }
        ]
      }
    },
    {
      "kind": "type_alias",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusPlugins"
      },
      "description": "A dynamic mapping of plugin ID to plugin status.",
      "type": {
        "kind": "instance_of",
        "type": {
          "namespace": "_builtins",
          "name": "Dictionary"
        },
        "generics": [
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValue"
            }
          }
        ]
      }
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValue"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "detail",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "documentationUrl",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": false
        },
        {
          "name": "level",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValueLevel"
            }
          },
          "required": true
        },
        {
          "name": "meta",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValueMeta"
            }
          },
          "required": true
        },
        {
          "name": "summary",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValueLevel"
      },
      "description": "Service status levels as human and machine readable values.",
      "isOpen": false,
      "members": [
        {
          "name": "available"
        },
        {
          "name": "degraded"
        },
        {
          "name": "unavailable"
        },
        {
          "name": "critical"
        }
      ]
    },
    {
      "kind": "type_alias",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseStatusPluginsValueMeta"
      },
      "description": "An unstructured set of extra metadata about this service.",
      "type": {
        "kind": "instance_of",
        "type": {
          "namespace": "_builtins",
          "name": "Dictionary"
        },
        "generics": [
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "UserDefined"
            }
          }
        ]
      }
    },
    {
      "kind": "interface",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseVersion"
      },
      "behaviors": [
        {
          "type": {
            "namespace": "_spec_utils",
            "name": "AdditionalProperties"
          },
          "generics": [
            {
              "kind": "instance_of",
              "type": {
                "namespace": "_builtins",
                "name": "UserDefined"
              }
            }
          ]
        }
      ],
      "properties": [
        {
          "name": "build_date",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        },
        {
          "name": "build_flavor",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_global",
              "name": "KibanaHttpApIsCoreStatusResponseVersionBuildFlavor"
            }
          },
          "required": true
        },
        {
          "name": "build_hash",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        },
        {
          "name": "build_number",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "double"
            }
          },
          "required": true
        },
        {
          "name": "build_snapshot",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "boolean"
            }
          },
          "required": true
        },
        {
          "name": "number",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "string"
            }
          },
          "required": true
        }
      ]
    },
    {
      "kind": "enum",
      "name": {
        "namespace": "_global",
        "name": "KibanaHttpApIsCoreStatusResponseVersionBuildFlavor"
      },
      "description": "The build flavour determines configuration and behavior of Kibana. On premise users will almost always run the \"traditional\" flavour, while other flavours are reserved for Elastic-specific use cases.",
      "isOpen": false,
      "members": [
        {
          "name": "serverless"
        },
        {
          "name": "traditional"
        }
      ]
    },
    {
      "kind": "request",
      "name": {
        "namespace": "_global",
        "name": "SystemStatusRequest"
      },
      "inherits": null,
      "path": [],
      "query": [
        {
          "name": "v7format",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "boolean"
            }
          },
          "required": false,
          "description": "Set to \"true\" to get the response in v7 format."
        },
        {
          "name": "v8format",
          "type": {
            "kind": "instance_of",
            "type": {
              "namespace": "_builtins",
              "name": "boolean"
            }
          },
          "required": false,
          "description": "Set to \"true\" to get the response in v8 format."
        }
      ],
      "body": {
        "kind": "no_body"
      }
    }
  ]
}
